### PR TITLE
Remove deprecated methods.

### LINKED ISSUE #1630: Dep warnings result in inference failure
From running JET:

```
  ││││││┌ limited_tendency!(Yₜ::ClimaCore.Fields.FieldVector{…}, Y::ClimaCore.Fields.FieldVector{…}, p::ClimaAtmos.AtmosCache{…}, t::Float64) @ ClimaAtmos /Users/charliekawczynski/Dropbox/Caltech/work/dev/CliMA/ClimaAtmos.jl/src/prognostic_equations/limited_tendencies.jl:4
  │││││││┌ tracer_hyperdiffusion_tendency!(Yₜ::ClimaCore.Fields.FieldVector{…}, Y::ClimaCore.Fields.FieldVector{…}, p::ClimaAtmos.AtmosCache{…}, t::Float64) @ ClimaAtmos /Users/charliekawczynski/Dropbox/Caltech/work/dev/CliMA/ClimaAtmos.jl/src/prognostic_equations/hyperdiffusion.jl:199
  ││││││││┌ node_horizontal_length_scale(space::ClimaCore.Spaces.SpectralElementSpace2D{…}) @ ClimaCore.Spaces /Users/charliekawczynski/.julia/packages/ClimaCore/vUG1Z/src/Spaces/spectralelement.jl:229
  │││││││││┌ getproperty(space::ClimaCore.Spaces.SpectralElementSpace2D{…}, name::Symbol) @ ClimaCore.Spaces /Users/charliekawczynski/.julia/packages/ClimaCore/vUG1Z/src/Spaces/spectralelement.jl:129
  ││││││││││┌ depwarn(msg::String, funcsym::Symbol) @ Base ./deprecated.jl:121
  │││││││││││┌ depwarn(msg::String, funcsym::Symbol; force::Bool) @ Base ./deprecated.jl:132
  ││││││││││││┌ firstcaller(bt::Vector{…}, funcsym::Symbol) @ Base ./deprecated.jl:152
  │││││││││││││┌ firstcaller(bt::Vector{…}, funcsyms::Tuple{…}) @ Base ./deprecated.jl:157
  ││││││││││││││┌ lookup(pointer::Ptr{…}) @ Base.StackTraces ./stacktraces.jl:193
  │││││││││││││││┌ get_inlinetable(mi::Core.MethodInstance) @ Base.StackTraces ./stacktraces.jl:104
  ││││││││││││││││ runtime dispatch detected: (%28::Any).linetable::Any
  │││││││││││││││└────────────────────
  │││││││││││││││┌ get_inlinetable(mi::Core.MethodInstance) @ Base.StackTraces ./stacktraces.jl:105
  ││││││││││││││││ runtime dispatch detected: Base.StackTraces.filter!(#1, %29::Any)::Any
  │
```

I think it'd be good if we take some time to remove these deprecations and do a breaking release.

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR removes a handful of deprecated methods. Since the next release will be a breaking release, we might as well rip off the band aid and remove some of the cruft.

Closes #1630.

### CODE REVIEW

**Comment by Sbozzolo** [comment]
Looks good, but I think it would be useful to test some downstream packages before merging this

File: src/Operators/numericalflux.jl [hunk 1]
```diff
@@ -102,14 +102,15 @@ function add_numerical_flux_boundary!(fn, dydt, args...)
     space = axes(dydt)
     Nq = Quadratures.degrees_of_freedom(Spaces.quadrature_style(space))
     topology = Spaces.topology(space)
+    boundary_surface_geometries = Spaces.grid(space).boundary_surface_geometries
 
     for (iboundary, boundarytag) in
         enumerate(Topologies.boundary_tags(topology))
```
[the remark this answers is not in the document]
Comment (charleskawczynski) [reply src/Operators/numericalflux.jl:108]: There's still a `boundary_tags` method (and a `boundary_names` method), and IIUC, the thing we pass in is actually the boundary names, not the tags, but both exist.

[the remark this answers is not in the document]
Comment (Sbozzolo) [reply src/Operators/numericalflux.jl:108]: Got it, thanks!

**Comment by Sbozzolo** [comment]
I have a PR in climaland to try this out and fix the issues (not many).
https://github.com/CliMA/ClimaLand.jl/pull/536

**Comment by charleskawczynski** [comment]
Buildkite passed [here](https://buildkite.com/clima/climacore-ci/builds/3229). Merging.

**Pull request merged**